apps/server/src/modules/authentication/uc/account.uc.ts

```diff
@@ -20,6 +33,112 @@ export class AccountUc {
 		private readonly permissionService: PermissionService
 	) {}
 
+	/**
+	 * This method processes the request on the GET account search endpoint from the account controller.
+	 *
+	 * @param currentUser the request user
+	 * @param query the request query
+	 * @throws {ValidationError}
+	 * @throws {ForbiddenOperationError}
+	 * @throws {EntityNotFoundError}
+	 */
+	async searchAccounts(currentUser: ICurrentUser, query: AccountSearchQueryParams): Promise<AccountSearchListResponse> {
+		const skip = query.skip ?? 0;
+		const limit = query.limit ?? 10;
+
+		if (skip < 0) {
+			throw new ValidationError('Skip is less than 0.');
```

Comment:
This could probably a DTO decorator, right? Or is that a problem because of the inheritance?

Reply:
Is it guaranteed that use cases will only be called from the controller or they be called form anywhere?